### ISSUE TITLE
tooltipField is not refreshed when cell is refreshed via cellRenderer refresh

### ISSUE THREAD

**Issue by catahoc** [ag-grid/ag-grid#1822]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
When you invoke gridApi.refreshCells(...), tooltip (defined via "tooltipField") is not refreshed sometimes - when cellRenderer/cellRendererFramework with "refresh" method implemented for this column

**Expected behavior**
When user invoke gridApi.refreshCells, all tooltips in refreshed cells should be refreshed even if renderers have refresh method implemented

**Minimal reproduction of the problem with instructions**
1. Make CellRenderer with refresh method
2. Make ColDef using this renderer and with defined "tooltipField"
3. Show row with this cell
4. Update tooltip for this cell
5. Invoke gridApi.refreshCells for this row/cell
6. Old tooltip is shown despite refresh was invoked

**What is the motivation / use case for changing the behavior?**
We have angular components as renderers. To improve refresh speed they all have "refresh(){...; return true}". Today I noticed, that tooltips are not refreshed

**Please tell us about your environment:**

Windows 7, Intellij Idea 2017.2.1, webpack dev server

* **ag-Grid version:** 12.0.2

* **Browser:**
Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36

* **Language:** all

**Comment by catahoc** [comment]
I have also tried to manually define 'title' in my cellRenderer's view. This works, but tooltip is shown only over "real" cellRenderer content: e.g. if cellRenderer is empty and shows "Required field" tooltip, user won't be able to see it.

**Comment by seanlandsman** [comment]
In your component refresh, do you update the tooltips there? 

From the docs here:

https://www.ag-grid.com/javascript-grid-cell-rendering-components/?framework=all#gsc.tab=0

// Mandatory - Get the cell to refresh. Return true if the refresh succeeded, otherwise return false.
// If you return false, the grid will remove the component from the DOM and create
// a new component in it's place with the new values.
refresh(params: any): boolean;

If you return true the grid will assume you've refreshed the cell - if false it will re-render the cell.

If this doesn't help we'll probably need a plunker or similar we can take a look at before we can assist you further

**Comment by catahoc** [comment]
In this scenario 'refresh' returns true.

But developer have no ability to modify tooltip in 'refresh' method: Column tooltip is something outside of custom cell component - just like "cellClassRules" which are updated regardless of 'refresh' return. This leads to situation where I can't both have cell tooltip and cell component with refresh: true. This is very disappointing.

And as I mentioned in previous comment, custom [title] inside component is not useful in ag-grid cells.

**Comment by makinggoodsoftware** [comment]
This is tracked by AG-1320 Check our pipeline https://www.ag-grid.com/ag-grid-pipeline/ for more information, thanks

**Comment by mohit000001** [comment]
@makinggoodsoftware , I have not found AG-1320 on given link. Please can share updated link if you have any.

**Comment by AG-Zoheil** [comment]
Hi

Thank you for your comment.

The comment you are referring to is form **six years ago**.

The grid has naturally advanced a lot since then. If you are still having problems, can we please ask you to open a new issue with a reproducible live scenario - you can use one of the plunker examples from our [documentation](https://ag-grid.com/javascript-data-grid/) as a starting point.

This would help our team support in the most efficient way, and also increase visibility for the wider AG Grid community in case of contributions and comments.

Much appreciated!

Kind regards,
Zoheil
